### PR TITLE
 Functional tests for .NET Core CLI 2.1.X

### PR DESCRIPTION
**Pull request by mikeharder**

### CODE REVIEW

File: .vsts/builds/e2e-tests.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Test.RuntimeIdentifier: win-x64
+      SelfContainedLinux:
+        Test.RuntimeIdentifier: linux-x64
+      SelfContainedMacOs:
+        Test.RuntimeIdentifier: osx-x64
+  steps:
+  - task: NodeTool@0
+    displayName: Install Node 10.x
+    inputs:
+      versionSpec: 10.x
```
Comment (natemcmaster) [inline .vsts/builds/e2e-tests.yml]: Should this be 8.x? (per issues in templates that we won't fix)

File: test/Cli.FunctionalTests/run-tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+param(
+    [switch]$ci,
+    $AssetRootUrl = $env:PB_ASSETROOTURL,
+    $AccessTokenSuffix = $env:PB_ACCESSTOKENSUFFIX,
+    $RestoreSources = $env:PB_RESTORESOURCES,
+    [ValidateSet('none', 'osx-x64', 'linux-x64', 'win-x64')]
+    $TestRuntimeIdentifier,
+    $HostRid,
+    $ProdConManifestUrl,
+    $ProcConChannel = 'release/2.2'
```
Comment (natemcmaster) [inline test/Cli.FunctionalTests/run-tests.ps1]: "release/2.1" would be a better default here.

Comment (natemcmaster) [reply test/Cli.FunctionalTests/run-tests.ps1]: Pro**d**ConChannel would be the right name to use here.

**Comment by natemcmaster** [comment]
FYI @mikeharder I cherry-picked a fix from 2.2 onto this PR to fix one of the mismatched prodcon variables.

File: .vsts/builds/e2e-tests.yml (hunk 1 above)
Comment (mikeharder) [reply .vsts/builds/e2e-tests.yml]: For now we can just test Node 10 on Windows.  I will create another PR to test both Node 8 and 10 on Windows.  Our 2.1 templates are only compatible with Node 8 on Mac.

**Pull request merged**